.github/workflows/review.yml

```diff
@@ -6,21 +6,26 @@ on:
   workflow_dispatch:
 
 jobs:
+  load-build-group:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+    - name: Checkout ${{ github.ref }}
+      uses: actions/checkout@v2
+
+      # build with the lowest versions of the OSs supported.
+      # ensures the glibc/musl versions are the oldest/most compatible.
+    - name: Load build group data
+      id: set-matrix
+      run: .github/scripts/matrix-from-json.sh .github/config/build-group.json
+
   build-group-test:
     runs-on: ubuntu-latest 
+    needs: load-build-group
     strategy:
       fail-fast: false
-      # build with the lowest versions of the OSs supported.
```

Comment:
Just wondering about the reason why we factor this out into a separate json file. It seems like the only json file (matrix config) which is used at two places is `build-group.json`. But this refactoring comes at the expense of:
- less clear workflow files (it is harder to quickly see what is going to run with this intermediate step)
- adding an additional `load-build-group` job, which both "pollutes" the workflow file as well as the logs of the Github action (as they show up as an individual box, even though this job is not really doing anything but reading the job configuration)
- instead of copy-pasting the one matrix once, we are now copy-pasting the `load-build-group` job into all workflows files

From the above, it does not seem like a good trade-off, thus I am wondering what other benefits the new method of storing the configuration matrices has?

Reply:
Generally, separation of concerns allows us to tell maintainers with no working knowledge of how the workflows are built (which includes me once I moved to the next task) that configuration of new node versions is to be done *without touching any workflow*. That's the main benefit.

Specifically, the build group is used three times, two here, once in release. For reason of maintainers minimal memory capacity (which includes me on most days) it has to either be all `YAML` or all `config/`. The latter is better then the former.

And finally reality - I just did the configuration needed for node 16. Without this specific implementation I'd go nuts...